### DISCUSSION TITLE
Obtain the misorientation angle of HCP?

### DISCUSSION THREAD

**Discussion by Brian-Liu-rx** [mtex-toolbox/mtex#2007] [Ask Anything]
Hi, everyone,
   I want to calculate the misorientation angle of two orientations for HCP metal. I know for FCC we can use the function "angle(ori1,ori2,CS,SS)" , where:
   CS = crystalSymmetry('m-3m', [4 4 4], 'mineral', 'Aluminum'); 
   SS = specimenSymmetry('1');  

   But for HCP, what is the CS and SS?

   Thanks!
   Brian Liu

**Comment by kilir** [comment]
Hi,
CS and SS depend on your material. However, your question is a bit unclear. a) `angle()` only gives you the misorientaion angle, not the misorientation b) actually you should/could use the symmetry of one your orientations `ori1` or `ori2` if both are not the same. If they are the same, no need to supply CS at all.
Hope that helps.
Cheers,
Rüdiger

**Reply by Brian-Liu-rx** [reply]
Hi Rüdiger,
   What I know is for FCC, we should consider the crystal symmetry, which means 24 kinds of equivalent variants. Then function "angle" can consider this by giving the CS and SS.
   But for HCP, such as Mg, what can I do to calculate the misorientation angle of two groups of euler angles? 

   Thanks!
    Ruxue Liu

**Reply by kilir** [reply]
No, `orientation/angle` always considers symmetry unless explicitly told not to. It takes the symmetry from the orientations, there is no need to supply it separately, actually it is even not considered as an argument. Also, if you deal with orientations, crystal symmetry always matters (except in the case of '1').

Cheers,Rüdiger

**Reply by Brian-Liu-rx** [reply]
Hi, Rüdiger,
I understand.
Here is the code for calculating the angle of two orientations for Mg rolling plates.
Could you please help me check whether it is correct?
Thansk!
Brian
```
clear
CS = crystalSymmetry('6/mmm', [3.2093 3.2093 5.2103], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Mg');
SS = specimenSymmetry('1'); 
ori1 = orientation.byEuler(0*degree,80*degree,0*degree,CS,SS);
ori2 = orientation.byEuler(0*degree,0*degree,0*degree,CS,SS);
angle(ori1,ori2)/degree
```

(The anwser is 80)

**Reply by kilir** [reply]
The answer is pretty much correct.

**Reply by Brian-Liu-rx** [reply]
Thanks! Rüdiger.
